test/framework/cluster_proxy.go

```diff
@@ -247,20 +251,60 @@ func (p *clusterProxy) GetCache(ctx context.Context) cache.Cache {
 	return p.cache
 }
 
-// Apply wraps `kubectl apply ...` and prints the output so we can see what gets applied to the cluster.
-func (p *clusterProxy) Apply(ctx context.Context, resources []byte, args ...string) error {
-	Expect(ctx).NotTo(BeNil(), "ctx is required for Apply")
-	Expect(resources).NotTo(BeNil(), "resources is required for Apply")
-
-	if err := testexec.KubectlApply(ctx, p.kubeconfigPath, resources, args...); err != nil {
-		var exitErr *exec.ExitError
-		if errors.As(err, &exitErr) {
-			return pkgerrors.New(fmt.Sprintf("%s: stderr: %s", err.Error(), exitErr.Stderr))
+// Creates or updates objects using the clusterProxy client.
+func (p *clusterProxy) CreateOrUpdate(ctx context.Context, resources []byte, selector ...string) error {
```

Comment:
being able to provide `selector ...string` but only at the end allowing to pass a single one just to make the arg optional looks a bit weird and unclean to me.

We could do it similar to `GetWorkloadCluster` and be able to pass in some `opts ...CreateOrUpdateOption`, which would also make it somewhat extendable if required in future.

Reply:
agreed, i wasnt too happy with this -- sounds good

Reply:
👋 @chrischdi pushed up some changes here, i think this covers all your feedback.  lmk if the implementation of options follows what you had in mind